popils_constants.py

```diff
@@ -9,12 +9,25 @@
 PRINCESS = (250, 20, 150)  # pink
 SUPPORT = (255, 255, 255)  # white
 BACKGROUND = (0, 0, 0)  # black
+COLORS = [PLAYER, LADDER, HARD, SOFT, PRINCESS, SUPPORT]
 
 ROW = 1
 COL = 0
 
+WINDOW_WIDTH = 500
+WINDOW_HEIGHT = 1000
 
-class VariableState(Enum):
-    UNUSED = 0
-    NEGATED = -1
-    NONNEGATED = 1
+GADGET_UNUSED = [LADDER, SUPPORT, LADDER, SUPPORT]
+GADGET_NEGATED = [LADDER, LADDER, SUPPORT, LADDER]
+GADGET_NONNEGATED = [SUPPORT, LADDER, LADDER, SUPPORT]
+SUB_GADGETS = [GADGET_UNUSED, GADGET_NEGATED, GADGET_NONNEGATED]
+GADGET_HEIGHT = 4  # in block units
```

Comment:
```suggestion
SUB_GADGET_HEIGHT = 4  # in block units
```
recommend prepending SUB as the full gadget height would be 6 blocks